src/quantum_object.jl

```diff
@@ -741,7 +742,17 @@ LinearAlgebra.rmul!(B::QuantumObject{<:AbstractArray}, a::Number) = (rmul!(B.dat
 @inline LinearAlgebra.mul!(y::AbstractVector{Ty}, A::QuantumObject{<:AbstractMatrix{Ta}}, x, α, β) where {Ty,Ta} =
     mul!(y, A.data, x, α, β)
 
-LinearAlgebra.sqrt(A::QuantumObject{<:AbstractArray{T}}) where {T} = QuantumObject(sqrt(A.data), A.type, A.dims)
+LinearAlgebra.sqrt(A::QuantumObject{<:AbstractArray{T}}) where {T} =
+    QuantumObject(sqrt(sparse_to_dense(A.data)), A.type, A.dims)
```

Comment:
We need `sparse_to_dense` because `sqrt` is not supported for sparse?

Reply:
Yes, I tried applying it on sparse matrices directly. But because sqrt uses eigen, which is not supported for sparse matrices.